components/calendar/CalendarWeekly.tsx

```diff
@@ -0,0 +1,410 @@
+                <div>
+                  <div className="sticky left-0 z-20 -ml-14 -mt-2.5 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
+                    7PM
+                  </div>
+                </div>
+                <div />
+                <div>
+                  <div className="sticky left-0 z-20 -ml-14 -mt-2.5 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
+                    8PM
+                  </div>
+                </div>
+                <div />
+                <div>
+                  <div className="sticky left-0 z-20 -ml-14 -mt-2.5 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
+                    9PM
+                  </div>
+                </div>
+                <div />
+                <div>
+                  <div className="sticky left-0 z-20 -ml-14 -mt-2.5 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
+                    10PM
+                  </div>
+                </div>
+                <div />
+                <div>
+                  <div className="sticky left-0 z-20 -ml-14 -mt-2.5 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
+                    11PM
+                  </div>
+                </div>
+                <div />
+              </div>
+
+              {/* Vertical lines */}
+              <div className="col-start-1 col-end-2 row-start-1 hidden grid-cols-7 grid-rows-1 divide-x divide-white sm:grid sm:grid-cols-7">
+                <div
+                  className={`col-start-1 row-span-full ${todayRow === 0 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-2 row-span-full ${todayRow === 1 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-3 row-span-full ${todayRow === 2 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-4 row-span-full ${todayRow === 3 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-5 row-span-full ${todayRow === 4 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-6 row-span-full ${todayRow === 5 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-7 row-span-full ${todayRow === 6 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+                <div
+                  className={`col-start-8 row-span-full w-8 ${todayRow === 7 && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
+                />
+              </div>
```

Comment:
@anna-murphy With this updated code, the cal events stack at the beginning of the week instead of at their proper locations.

```
  const [week, setWeek] = useState<Array<weekProps>>([])

  useEffect(() => {
    // Generate unique IDs for list items
    const updatedWeeks = Array.from(Array(8).keys()).map((_, i) => ({
      id: self.crypto.randomUUID(),
      num: i,
    }))
    setWeek(updatedWeeks)
  }, [])


                {week.map(function (week) {
                  return (
                    <div
                      key={week.id}
                      className={`row-span-full col-start-${week.num + 1} ${
                        week.num === 7 ? "w-8" : ""
                      } ${week.num === todayRow && isSameDay(activeDate, currentDate) ? "bg-secondary-yellow-100" : ""}`}
                    />
                  )
                })}
```

Reply:
<img width="832" alt="Screenshot 2024-10-15 at 2 20 55 PM" src="https://github.com/user-attachments/assets/8cd5f8ca-f694-458d-9cd6-ada4d855f4a5">